### ISSUE TITLE
report a fatal scene in High concurrency

### ISSUE THREAD

**Issue by dongkui0712** [sirupsen/logrus#806]
```
fatal error: concurrent map read and map write

goroutine 366 [running]:
runtime.throw(0xe4db0b, 0x21)
	/usr/local/Cellar/go/1.10.1/libexec/src/runtime/panic.go:616 +0x81 fp=0xc42191d200 sp=0xc42191d1e0 pc=0x429531
runtime.mapaccess1_faststr(0xd031c0, 0xc420272360, 0xe3505d, 0x4, 0xca6520)
	/usr/local/Cellar/go/1.10.1/libexec/src/runtime/hashmap_fast.go:181 +0x421 fp=0xc42191d270 sp=0xc42191d200 pc=0x40a341
projects/FlashPay_go/vendor/github.com/sirupsen/logrus.(*TextFormatter).Format(0xc4203c4340, 0xc422c740a0, 0x70, 0xd727c0, 0xc420eef200, 0xc4203a89b0, 0xc42191d440)
	~/go/src/projects/FlashPay_go/vendor/github.com/sirupsen/logrus/text_formatter.go:120 +0x415 fp=0xc42191d3f0 sp=0xc42191d270 pc=0x8d48f5
projects/FlashPay_go/vendor/github.com/sirupsen/logrus.(*Entry).write(0xc422c740a0)
	~/go/src/projects/FlashPay_go/vendor/github.com/sirupsen/logrus/entry.go:140 +0x51 fp=0xc42191d498 sp=0xc42191d3f0 pc=0x8d0d71
projects/FlashPay_go/vendor/github.com/sirupsen/logrus.Entry.log(0xc420394410, 0xc420272360, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x2, ...)
	~/go/src/projects/FlashPay_go/vendor/github.com/sirupsen/logrus/entry.go:118 +0x1b9 fp=0xc42191d508 sp=0xc42191d498 pc=0x8d0aa9
projects/FlashPay_go/vendor/github.com/sirupsen/logrus.(*Entry).Error(0xc42021e230, 0xc42191d5b8, 0x1, 0x1)
	~/go/src/projects/FlashPay_go/vendor/github.com/sirupsen/logrus/entry.go:181 +0xac fp=0xc42191d578 sp=0xc42191d508 pc=0x8d131c
projects/FlashPay_go/vendor/github.com/sirupsen/logrus.(*Entry).Errorln(0xc42021e230, 0xc42191d658, 0x1, 0x1)
	~/go/src/projects/FlashPay_go/vendor/github.com/sirupsen/logrus/entry.go:276 +0xd6 fp=0xc42191d5d8 sp=0xc42191d578 pc=0x8d2086
```

**Comment by dgsb** [comment]
@macnibblet which version of logrus are you using ?
We used to have a concurrency bug on the field map which have been solved by the use of a mutex.
If your still have concurrency error, please open a new issue.

**Comment by macnibblet** [comment]
```[[projects]]
  name = "github.com/sirupsen/logrus"
  packages = ["."]
  revision = "3e01752db0189b9157070a0e1668a620f9a85da2"
  version = "v1.0.6"
```

One of the crashes, this is the bulk processor also have another one which is strictly field operations
```goroutine 27 [running]:
runtime.throw(0xe88c0c, 0x26)
	/usr/local/go/src/runtime/panic.go:616 +0x81 fp=0xc420d17968 sp=0xc420d17948 pc=0x429c61
runtime.mapiternext(0xc420b48660)
	/usr/local/go/src/runtime/hashmap.go:747 +0x55c fp=0xc420d179f8 sp=0xc420d17968 pc=0x40873c
reflect.mapiternext(0xc420b48660)
	/usr/local/go/src/runtime/hashmap.go:1223 +0x2b fp=0xc420d17a10 sp=0xc420d179f8 pc=0x40985b
reflect.Value.MapKeys(0xd2cb40, 0xc420fbf110, 0x15, 0x0, 0x407322, 0xc420d17ba0)
	/usr/local/go/src/reflect/value.go:1134 +0x13f fp=0xc420d17ab8 sp=0xc420d17a10 pc=0x4ad6ff
encoding/json.(*mapEncoder).encode(0xc4205840b0, 0xc421ef4fd0, 0xd2cb40, 0xc420fbf110, 0x15, 0xd20100)
	/usr/local/go/src/encoding/json/encode.go:668 +0xad fp=0xc420d17c18 sp=0xc420d17ab8 pc=0x4ec55d
encoding/json.(*mapEncoder).(encoding/json.encode)-fm(0xc421ef4fd0, 0xd2cb40, 0xc420fbf110, 0x15, 0xc420fb0100)
	/usr/local/go/src/encoding/json/encode.go:700 +0x64 fp=0xc420d17c58 sp=0xc420d17c18 pc=0x4f6d44
encoding/json.(*encodeState).reflectValue(0xc421ef4fd0, 0xd2cb40, 0xc420fbf110, 0x15, 0x100)
	/usr/local/go/src/encoding/json/encode.go:325 +0x82 fp=0xc420d17c90 sp=0xc420d17c58 pc=0x4e9f82
encoding/json.(*encodeState).marshal(0xc421ef4fd0, 0xd2cb40, 0xc420fbf110, 0xc4202e0100, 0x0, 0x0)
	/usr/local/go/src/encoding/json/encode.go:298 +0xa5 fp=0xc420d17cc8 sp=0xc420d17c90 pc=0x4e9c75
encoding/json.Marshal(0xd2cb40, 0xc420fbf110, 0x36, 0x40, 0xc4202e2b40, 0x36, 0x0)
	/usr/local/go/src/encoding/json/encode.go:161 +0x5f fp=0xc420d17d10 sp=0xc420d17cc8 pc=0x4e984f
git.interactivesolutions.se/epulze/tournaments/vendor/github.com/olivere/elastic.(*BulkIndexRequest).Source(0xc422424d00, 0x0, 0x0, 0x0, 0xc4204120c0, 0x0)
	/go/src/git.interactivesolutions.se/epulze/tournaments/vendor/github.com/olivere/elastic/bulk_index_request.go:219 +0x3e3 fp=0xc420d17e80 sp=0xc420d17d10 pc=0x85ae73
git.interactivesolutions.se/epulze/tournaments/vendor/github.com/olivere/elastic.(*bulkWorker).work(0xc42002e600, 0xf88ee0, 0xc4203bb3c0)
	/go/src/git.interactivesolutions.se/epulze/tournaments/vendor/github.com/olivere/elastic/bulk_processor.go:454 +0x3b3 fp=0xc420d17fc8 sp=0xc420d17e80 pc=0x85df63
runtime.goexit()
	/usr/local/go/src/runtime/asm_amd64.s:2361 +0x1 fp=0xc420d17fd0 sp=0xc420d17fc8 pc=0x455fe1
created by git.interactivesolutions.se/epulze/tournaments/vendor/github.com/olivere/elastic.(*BulkProcessor).Start
	/go/src/git.interactivesolutions.se/epulze/tournaments/vendor/github.com/olivere/elastic/bulk_processor.go:306 +0x1d7

```

**Comment by dgsb** [comment]
@macnibblet I don't think this trace is related to logrus, we can not help with this error.
But for your information, the latest tagged logrus version is the 1.2.0.
I suggest you switch to this one, a race condition has been fixed since v1.1.0.
